src/promptflow/promptflow/azure/_utils/gerneral.py

```diff
@@ -3,14 +3,69 @@
 # ---------------------------------------------------------
 
 import jwt
+from promptflow.exceptions import ValidationException
 
 
 def is_arm_id(obj) -> bool:
     return isinstance(obj, str) and obj.startswith("azureml://")
 
 
+def get_aml_token(credential):
+    from azure.ai.ml._azure_environments import _get_aml_resource_id_from_metadata
+    from azure.ai.ml._azure_environments import _resource_to_scopes
+
+    # these code copied from _set_headers_with_user_aml_token function of azure.ai.ml.operations._job_operations
+    aml_resource_id = _get_aml_resource_id_from_metadata()
+    azure_ml_scopes = _resource_to_scopes(aml_resource_id)
+    token = credential.get_token(*azure_ml_scopes).token
+    # validate token has aml audience
+    decoded_token = jwt.decode(
+        token,
+        options={"verify_signature": False, "verify_aud": False},
+    )
+    if decoded_token.get("aud") != aml_resource_id:
+        msg = """AAD token with aml scope could not be fetched using the credentials being used.
+        Please validate if token with {0} scope can be fetched using credentials provided to MLClient.
+        Token with {0} scope can be fetched using credentials.get_token({0})
+        """
+        raise ValidationException(
+            message=msg.format(*azure_ml_scopes),
+        )
+
+    return token
+
+
+def get_base_token(credential):
```

Comment:
let's call this get_arm_token

management.azure.com is the rest endpoint for Azure Resource Manager

Reply:
done